src/osvr/Server/ConfigureServerFromFile.cpp

```diff
@@ -0,0 +1,181 @@
+            }
+        }
+
+        {
+            log->info() << "Instantiating configured drivers...";
+            bool success = srvConfig.instantiateDrivers();
+            if (!srvConfig.getSuccessfulInstantiations().empty()) {
+                log->info() << "Successes:";
+                for (auto const &driver :
+                    srvConfig.getSuccessfulInstantiations()) {
+                    log->info() << " - " << driver;
+                }
+            }
+            if (!srvConfig.getFailedInstantiations().empty()) {
+                log->error() << "Errors:";
+                for (auto const &error : srvConfig.getFailedInstantiations()) {
+                    log->error() << " - " << error.first << "\t"
+                        << error.second;
+                }
+            }
+        }
+
+        if (srvConfig.processExternalDevices()) {
+            log->info()
+                << "External devices found and parsed from config file.";
+        }
+
+        if (srvConfig.processRoutes()) {
+            log->info() << "Routes found and parsed from config file.";
+        }
+
+        if (srvConfig.processAliases()) {
+            log->info() << "Aliases found and parsed from config file.";
+        }
+
+        if (srvConfig.processDisplay()) {
+            log->info()
+                << "Display descriptor found and parsed from config file.";
+        } else {
+            log->info()
+                << "Using OSVR HDK for display configuration. "
+                "Did not find an alternate valid 'display' object in config "
+                "file.";
+        }
+
+        if (srvConfig.processRenderManagerParameters()) {
+            log->info() << "RenderManager config found and parsed from the "
+                "config file.";
+        }
+
+        log->info() << "Triggering automatic hardware detection...";
+        ret->triggerHardwareDetect();
+
+        return ret;
+    }
+
+    ServerPtr configureServerFromFile(std::string const &configName) {
+        auto log =
+            ::osvr::util::log::make_logger(::osvr::util::log::OSVR_SERVER_LOG);
+
+        ServerPtr ret;
+        log->info() << "Using config file '" << configName << "'.";
+        std::ifstream config(configName);
+        if (!config.good()) {
+            log->error() << "Could not open config file!";
+            log->error() << "Searched in the current directory; file may be "
+                "misspelled, missing, or in a different directory.";
+            return nullptr;
+        }
+
+        std::stringstream sstr;
+        sstr << config.rdbuf();
+        return configureServerFromString(sstr.str());
+    }
+
+    ServerPtr configureServerFromFirstFileInList(
+        std::vector<std::string> const &configNames) {
+        auto log =
+            ::osvr::util::log::make_logger(::osvr::util::log::OSVR_SERVER_LOG);
+
+        for (const auto name : configNames) {
+            std::ifstream config(name);
+            if (config.good()) {
```

Comment:
It's logged in `configureServerFromFile`, but I adjusted the log there to be more consistent with the others in wording.